### PR TITLE
[SOL-2197] Do not allow unsupported CCs from new basket page.

### PR DESCRIPTION
**Pull request by iivic**
@mjfrey @vkaracic @marjev https://openedx.atlassian.net/browse/SOL-2197
The following credit card types are supported: American express, discover, mastercard, visa.
I also included a card number check before the card image is displayed that means the image will not be displayed if the card number doesn't pass the validation.
For unsupported card types a message will appear
screen shot 2017-01-27 at 13 29 56

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @iivic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?number=1113&repo=edx%2Fecommerce"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

**Review by marjev — APPROVED** [review]

**Review by vkaracic — APPROVED** [review]
I guess you could it do this way. But you could also just remove the unsupported credit card types from here: https://github.com/edx/ecommerce/blob/49e89aa3af5ff0c928d3a4c1c5ac23da52c0de53/ecommerce/static/js/utils/credit_card.js#L31
That way we don't have code that's not used anywhere.

File: ecommerce/static/js/pages/basket_page.js [hunk 1]
```diff
@@ -124,7 +124,7 @@ define([
 
             if (!CreditCardUtils.isValidCardNumber(cardNumber)) {
                 appendCardValidationErrorMsg(event, cardNumberField, 'Invalid card number');
-            } else if (typeof cardType === 'undefined') {
+            } else if (typeof cardType === 'undefined' || !isCardTypeSupported(cardType.name)) {
                 appendCardValidationErrorMsg(event, cardNumberField, 'Unsupported card type');
```
Comment (mjfrey) [inline ecommerce/static/js/pages/basket_page.js:128]: we might want a better message like:
"edX accepts Visa, MasterCard, American Express, and Discover credit cards"

Comment (mjfrey) [reply ecommerce/static/js/pages/basket_page.js:128]: @iivic just leave "Unsupported card type"

**Review by mjfrey — APPROVED** [review]

**Pull request merged**